js/src/components/pmax-improve-assets-banner/index.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useDispatch } from '@wordpress/data';
+import { Notice } from '@wordpress/components';
+import { getHistory } from '@woocommerce/navigation';
+import { store as preferencesStore } from '@wordpress/preferences';
+
+/**
+ * Internal dependencies
+ */
+import {
+	CAMPAIGN_TYPE_PMAX,
+	PREFERENCES_STORE_NAMESPACE,
+	PMAX_IMPROVE_PERFORMANCE_MAX_AD_STRENGTH,
+	DAY_IN_SECONDS,
+} from '~/constants';
+import { getEditCampaignUrl } from '~/utils/urls';
+import AppButton from '~/components/app-button';
+import usePreference from '~/hooks/usePreference';
+import useAdsCampaigns from '~/hooks/useAdsCampaigns';
+import useAdsRecommendations from '~/hooks/useAdsRecommendations';
+import './index.scss';
+
+const PREFERENCE_BANNER_KEY = 'pmax-improve-assets-banner';
+
+/**
+ * Displays a dismissible banner prompting users to improve assets for their highest-spending enabled Performance Max (PMAX) campaign.
+ *
+ * The banner is shown only if:
+ * - The preference expiry is undefined or expired.
+ * - There are enabled PMAX campaigns.
+ * - There are relevant asset improvement recommendations.
+ *
+ * When dismissed, the banner will not reappear until the expiry time elapses.
+ * Clicking "Improve Assets" navigates to the asset group edit page for the highest-spending PMAX campaign.
+ *
+ * @return {JSX.Element|null} The banner component, or null if not applicable.
+ */
+const PMaxImproveAssetsBanner = () => {
+	const { set } = useDispatch( preferencesStore );
+	const { expiry } = usePreference( PREFERENCE_BANNER_KEY ) || {};
+	const { data: adsCampaignsData } = useAdsCampaigns();
+	const { recommendations } = useAdsRecommendations(
+		PMAX_IMPROVE_PERFORMANCE_MAX_AD_STRENGTH
+	);
+
+	if ( expiry !== undefined ) {
+		if ( Date.now() < expiry ) {
+			// Do not render the banner if not expired
+			return null;
+		}
+
+		// If expired, reset the preference
+		set( PREFERENCES_STORE_NAMESPACE, PREFERENCE_BANNER_KEY, {
+			expiry: undefined, // Reset to undefined to show the banner again
+		} );
```

Comment:
Should this be in a `useEffect()` hook to avoid unnecessary re-renders?

Reply:
Yes, just updated it. When I tested there were no re-renders but just to make sure, I've used `useEffect`